src/ModularPipelines/Exceptions/PipelineException.cs

```diff
@@ -16,6 +16,9 @@ public PipelineException(string? message, Exception? innerException) : base(mess
     {
     }
 
+#if NET8_0_OR_GREATER
+    [Obsolete(DiagnosticId = "SYSLIB0051")] // add this attribute to the serialization ctor
+#endif
     protected PipelineException(SerializationInfo info, StreamingContext context) : base(info, context)
```

Comment:
Actually, I don't think this constructor is ever used. All exceptions that inherit it just use a basic message or inner exception constructor.

Can you check this please? And if so, just remove this constructor completely. I think that's cleaner than keeping it in if nothing is using it at current. And that'll prevent people from using it in future if it's a now obsolete piece of code

Reply:
Yep, works when removed.